### PR TITLE
use api to do jobpersistence query

### PR DESCRIPTION
**Pull request by xiaohansong**
## What
Convert JobPeristence dependency to airbyteApiCilent dependency; because normalization summary check is on sync workflow so they will also be executed by multicloud.

Converted Optional.of(recordsCommited) into two fields (hasRecordsCommited, recordsCommited).

I tried to fit everything back in all existing logic but would like a second.

### CODE REVIEW

File: airbyte-api/src/main/openapi/config.yaml [hunk 1]
```diff
@@ -4823,6 +4844,26 @@ components:
       properties:
         succeeded:
           type: boolean
+    AttemptNormalizationStatuses:
```
Comment (pmossman) [inline airbyte-api/src/main/openapi/config.yaml:4847]: nit: I think the convention in this file would be to name this `AttemptNormalizationStatusReadList` (see `WorkspaceReadList` as an example of the convention in this file)

File: airbyte-api/src/main/openapi/config.yaml [hunk 2]
```diff
@@ -4823,6 +4844,26 @@ components:
...
+      type: object
+      properties:
+        attemptNormalizationStatus:
```
Comment (pmossman) [inline airbyte-api/src/main/openapi/config.yaml:4850]: nit: would name this `attemptNormalizationStatuses`

File: airbyte-server/src/main/java/io/airbyte/server/converters/JobConverter.java [hunk 3]
```diff
@@ -240,4 +242,11 @@ public SynchronousJobRead getSynchronousJobRead(final SynchronousJobMetadata met
         .logs(getLogRead(metadata.getLogPath()));
   }
 
+  public static AttemptNormalizationStatusRead convertAttemptNormalizationStatus(
+                                                                                 AttemptNormalizationStatus databaseStatus) {
+    return new AttemptNormalizationStatusRead().attemptNumber(databaseStatus.attemptNumber())
```
Comment (pmossman) [inline airbyte-server/src/main/java/io/airbyte/server/converters/JobConverter.java:247]: nit: I'd put each property on its own line for readability:
```
    return new AttemptNormalizationStatusRead()
        .attemptNumber(databaseStatus.attemptNumber())
        .hasRecordsCommitted(!databaseStatus.recordsCommitted().isEmpty())        
        .recordsCommitted(databaseStatus.recordsCommitted().orElse(0L))
        .hasNormalizationFailed(databaseStatus.normalizationFailed());
```

File: airbyte-server/src/main/java/io/airbyte/server/handlers/JobHistoryHandler.java [hunk 4]
```diff
@@ -146,6 +147,12 @@ public List<JobRead> getLatestSyncJobsForConnections(final List<UUID> connection
         .collect(Collectors.toList());
   }
 
+  public AttemptNormalizationStatuses getAttemptNormalizationStatuses(final JobIdRequestBody jobIdRequestBody) throws IOException {
+    return new AttemptNormalizationStatuses()
+        .attemptNormalizationStatus(jobPersistence.getAttemptNormalizationStatusesForJob(jobIdRequestBody.getId()).stream()
```
Comment (pmossman) [inline airbyte-server/src/main/java/io/airbyte/server/handlers/JobHistoryHandler.java:152]: nit: I'd put the inside logic into its own variable to make it easier to read:

```
final var statuses = jobPersistence.getAttemptNormalizationStatusesForJob(jobIdRequestBody.getId())
    .stream()
    .map(JobConverter::convertAttemptNormalizationStatus)
    .collect(Collectors.toList());

**Review by pmossman — DISMISSED** [review]
Made a few naming and style suggestions. Logic looks good to me though I'd make sure @alovew signs off on the activity code before merging!

**Review by alovew — APPROVED** [review]

**Pull request merged**